decompiler/data/game_text.cpp

```diff
@@ -162,6 +163,28 @@ std::string write_game_text(
     result += ")\n\n";
   }
 
+  // add our own custom text additions from new_strings.jsonc
+  // - first add the strings that are the same across all languages
+  for (auto const& [key, val] : cfg.new_strings_same_across_langs) {
+    result += fmt::format("(#x{}\n  ", key);
+    for (int i = 0; i < languages.size(); i++) {
+      result += fmt::format("\"{}\"\n  ", val);
+    }
+    result += ")\n\n";
+  }
+  // - now add the ones that are different, if they do not have all languages defined, pad with
+  // placeholders
+  for (auto const& [key, val] : cfg.new_strings_different_across_langs) {
+    result += fmt::format("(#x{}\n  ", key);
+    for (auto const& str : val) {
+      result += fmt::format("\"{}\"\n  ", str);
+    }
+    for (int i = 0; i < languages.size() - val.size(); i++) {
+      result += fmt::format("\"{}\"\n  ", "TODO");
+    }
+    result += ")\n\n";
+  }
+
```

Comment:
this approach is fine for now, but in the future it might be nicer to have separate files for our custom text and the decompiled text.  We could modify the text tool to take multiple inputs and combine them.